proto/src/domain/blocks.rs

```diff
@@ -0,0 +1,133 @@
+            block_num: header.block_num(),
+            chain_root: Some(header.chain_root().into()),
+            account_root: Some(header.account_root().into()),
+            nullifier_root: Some(header.nullifier_root().into()),
+            note_root: Some(header.note_root().into()),
+            batch_root: Some(header.batch_root().into()),
+            proof_hash: Some(header.proof_hash().into()),
+            version: u64::from(header.version())
+                .try_into()
+                .expect("TODO: BlockHeader.version should be u64"),
+            timestamp: header.timestamp().into(),
+        }
+    }
+}
+
+impl TryFrom<&block_header::BlockHeader> for BlockHeader {
+    type Error = ParseError;
+
+    fn try_from(value: &block_header::BlockHeader) -> Result<Self, Self::Error> {
+        value.clone().try_into()
+    }
+}
+
+impl TryFrom<block_header::BlockHeader> for BlockHeader {
+    type Error = ParseError;
+
+    fn try_from(value: block_header::BlockHeader) -> Result<Self, Self::Error> {
+        Ok(BlockHeader::new(
+            value
+                .prev_hash
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(prev_hash)))?
+                .try_into()?,
+            value.block_num,
+            value
+                .chain_root
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(chain_root)))?
+                .try_into()?,
+            value
+                .account_root
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(account_root)))?
+                .try_into()?,
+            value
+                .nullifier_root
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(nullifier_root)))?
+                .try_into()?,
+            value
+                .note_root
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(note_root)))?
+                .try_into()?,
+            value
+                .batch_root
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(batch_root)))?
+                .try_into()?,
+            value
+                .proof_hash
+                .ok_or(block_header::BlockHeader::missing_field(stringify!(proof_hash)))?
+                .try_into()?,
+            value.version.into(),
+            value.timestamp.into(),
+        ))
+    }
+}
+
+// BLOCK INPUTS
+// ================================================================================================
+
+/// Information needed from the store to build a block
+#[derive(Clone, Debug)]
+pub struct BlockInputs {
+    /// Previous block header
+    pub block_header: BlockHeader,
+
+    /// MMR peaks for the current chain state
+    pub chain_peaks: MmrPeaks,
+
+    /// The hashes of the requested accounts and their authentication paths
+    pub account_states: Vec<AccountInputRecord>,
+
+    /// The requested nullifiers and their authentication paths
+    pub nullifiers: Vec<NullifierWitness>,
+}
```

Comment:
Since this struct is used only in the block producer, I would move it there (probably into `src/block.rs` file).

Also, I would probably change the `account_states` and `nullifiers` into maps because this is what is happening in the block witness constructor anyway. So, the struct would look something like this:
```Rust
pub struct BlockInputs {
    pub block_header: BlockHeader,
    pub chain_peaks: MmrPeaks,
    pub accounts: BTreeMap<AccountId, AccountWitness>,
    pub nullifiers: BTreeMap<Digest, SmtProof>,
}

pub struct AccountWitness {
    pub hash: Digest,
    pub proof: MerklePath,
}
```